tools/spectral/ipa/__tests__/eachResourceHasGetMethod.test.js

```diff
@@ -0,0 +1,86 @@
+import testRule from './__helpers__/testRule';
+import { DiagnosticSeverity } from '@stoplight/types';
+
+testRule('xgen-IPA-104-resource-has-GET', [
+  {
+    name: 'valid methods',
+    document: {
+      paths: {
+        '/standard': {
+          post: {},
+          get: {},
+        },
+        '/standard/{exampleId}': {
+          get: {},
+          patch: {},
+          delete: {},
+        },
+        '/custom': {
+          post: {},
+          get: {},
+        },
+        '/custom/{exampleId}': {
+          get: {},
+          patch: {},
+          delete: {},
+        },
+        '/custom:method': {
+          post: {},
+        },
+        '/singleton': {
+          get: {},
+        },
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid methods',
+    document: {
+      paths: {
+        '/standard': {
+          post: {},
+          get: {},
+        },
+        '/standard/{exampleId}': {
+          patch: {},
+          delete: {},
+        },
+        '/custom': {
+          post: {},
+          get: {},
+        },
+        '/custom/{exampleId}': {
+          patch: {},
+          delete: {},
+        },
+        '/custom:method': {
+          post: {},
+        },
+        '/singleton': {
+          patch: {},
+        },
+      },
+    },
+    errors: [
+      {
+        code: 'xgen-IPA-104-resource-has-GET',
```

Comment:
Q: Why does `/custom` cause an error if we are not checking `child` and `custom` methods? Same question for `/standard` as well

Reply:
Is it because we’re leaving all the checks to the parent, which is responsible for handling both child routes and custom methods?

Reply:
Yes, each "parent" path is considered the id for a resource, and the rule checks that the resource has a get (in children or if it's a singleton -> on itself). We don't require custom methods to have a GET, they can be either GET or POST (https://docs.devprod.prod.corp.mongodb.com/ipa/109), but in the case here `custom/{exampleId}` should have a GET, so we get an error for `/custom` path since it's the resource parent. Let me know if that makes sense!